### PR TITLE
Show overview of gateway status

### PR DESCRIPTION
**Pull request by AramAlsabti**
Supports fetching the status of all gateways. The backend now also subscribes to a topic on Chirpstack which receives [gateway status messages](https://www.chirpstack.io/gateway-bridge/payloads/states/#conn-gateway-connection) from time to time. It stores the status messages in a separate table which can then be fetched by a client

Note that this differs from the application server which shows gateways as offline if they haven't received a packages in the last couple of minutes.

https://github.com/OS2iot/OS2IoT-frontend/pull/84

### CODE REVIEW

File: src/services/chirpstack/chirpstack-gateway.service.ts [hunk 1]
```diff
@@ -112,6 +112,11 @@ export class ChirpstackGatewayService extends GenericChirpstackConfigurationServ
         };
     }
 
+    /**
+     * Fetch gateways individually. This gives us the tags which contain the OS2 organization id.
+     * This is a very expensive operation, but it's the only way to retrieve gateway tags.
```
Comment (GufCab) [inline src/services/chirpstack/chirpstack-gateway.service.ts:117]: If this turns out to be a problem, we will have to look into having a Gateway entity in the OS2iot database

File: src/entities/dto/kafka/raw-iot-device-request.dto.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { RawRequestDto } from "./raw-request.dto";
+
+export class RawIotDeviceRequestDto extends RawRequestDto {
```
Comment (GufCab) [inline src/entities/dto/kafka/raw-iot-device-request.dto.ts:3]: Nitpick: RawIotDeviceRequestDto -> RawIoTDeviceRequestDto

File: src/services/data-management/gateway-persistence.service.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        await this.deleteOldStatusHistories(dto.gatewayId);
+    }
+
+    private mapDtoToEntity(
+        dto: RawGatewayStateDto,
+        messageState: ChirpstackMQTTConnectionStateMessageDto
+    ) {
+        const statusHistory = new GatewayStatusHistory();
+        statusHistory.mac = dto.gatewayId;
+        statusHistory.timestamp = dto.unixTimestamp
```
Comment (GufCab) [inline src/services/data-management/gateway-persistence.service.ts:58]: Is this going to be an issue with UTC?

**Review by GufCab — CHANGES_REQUESTED** [review]
Very minor feedback, solve these and merge

File: src/services/chirpstack/chirpstack-gateway.service.ts (hunk 1 above)
Comment (AramAlsabti) [reply src/services/chirpstack/chirpstack-gateway.service.ts:117]: I agree.

File: src/services/data-management/gateway-persistence.service.ts (hunk 3 above)
Comment (AramAlsabti) [reply src/services/data-management/gateway-persistence.service.ts:58]: Currently, the timestamp is set and passed from [ReceiveDataService.buildMessage()](https://github.com/OS2iot/OS2IoT-backend/pull/170/files#diff-09adc96f1df326bcc3a378c68f40e9bc0bdcb3dcffd741262e1a14feb1748f37R58). The value is `new Date().valueOf()` which is always the unix time in ms.
When we store a Date object, they are converted to UTC time.

**Comment by AramAlsabti** [comment]
As discussed, the migration timestamp has been bumped. Remember to update the entry in your migrations table.

**Pull request merged**